src/domain/flow-system/services/tests/tests/utils/flow_harness_shared.rs

```diff
@@ -181,9 +181,8 @@ impl FlowHarness {
         }
     }
 
-    pub async fn create_root_dataset(&self, dataset_alias: DatasetAlias) -> DatasetID {
-        let result = self
-            .dataset_repo
+    pub async fn create_root_dataset(&self, dataset_alias: DatasetAlias) -> CreateDatasetResult {
```

Comment:
Optional: leave `create_root_dataset()` method as it is, but add a new one (to be used inside `create_root_dataset()` as well) that returns `CreateDatasetResult`.

Motivation, this way we can take away the pressure on the reader from ....
```rust
        .await
        .dataset_handle
        .id;
```
... since, only in one (if I counted correctly) case is `CreateDatasetResult` used.

Reply:
Changed